src/logic/transactionLogic.ts

```diff
@@ -2,29 +2,55 @@ import { v4 as uuidv4 } from 'uuid';
 import { getSplTransferById, createSplTransfer as createSplTransferInDb } from '@/services/db/queries/splTransfer';
 import { SplTransfer, transactionStatuses } from '@/app/types/splTransfer';
 import { supportedMints } from '@/app/config/mint';
-
+import { EmbeddedWallet, ix_TransferSPL } from '@/utils/EmbeddedWallet';
 export async function getSplTransfer(id: string): Promise<SplTransfer | null> {
   return await getSplTransferById(id);
 }
 
-export async function createSplTransfer(sender: string, destination: string, amount: string, mintSymbol: string): Promise<string | null> {
-  // TODO: get and set the current fee
-  const mint = supportedMints[mintSymbol as keyof typeof supportedMints];
+export async function createSplTransfer(sender: string, destination: string, amount: string, mintSymbol: string): Promise<SplTransfer> {
+  const mintAddress = supportedMints[mintSymbol as keyof typeof supportedMints];
+  const relayWallet = await EmbeddedWallet.get();
+
+  const RELAY_FEE = '500000'; // 0.50 USDC/USDT (6 decimal places)
+
+  const ix_fee = await ix_TransferSPL(
+    sender,
+    destination,
+    RELAY_FEE,
+    mintAddress,
+    mintAddress
+  );
+
+  const ix_transfer = await ix_TransferSPL(
+    sender,
+    destination,
```

Comment:
Do these need to be token accounts? I'm expecting the end user to pass in their main account and we'd have to derive the correct token acct

Reply:
Yes, that is [the current logic](https://github.com/kilogold/tinywallet/blob/sprint-edge/src/instructionbuilder.ts).

Reply:
Got it. if the destination token account doesn't exist this will fail on chain right? we need to add a create account instruction?